Assets/Scripts/Infrastructure/NetworkObjectPool.cs

```diff
@@ -141,54 +165,40 @@ private GameObject CreateInstance(GameObject prefab)
         /// <param name="position"></param>
         /// <param name="rotation"></param>
         /// <returns></returns>
-        private NetworkObject GetNetworkObjectInternal(GameObject prefab, Vector3 position, Quaternion rotation)
+        NetworkObject GetNetworkObjectInternal(GameObject prefab, Vector3 position, Quaternion rotation)
         {
-            var queue = pooledObjects[prefab];
-
-            NetworkObject networkObject;
-            if (queue.Count > 0)
-            {
-                networkObject = queue.Dequeue();
-            }
-            else
-            {
-                networkObject = CreateInstance(prefab).GetComponent<NetworkObject>();
-            }
-
-            // Here we must reverse the logic in ReturnNetworkObject.
-            var go = networkObject.gameObject;
-            go.SetActive(true);
+            var networkObject = m_PooledObjects[prefab].Get();
 
-            go.transform.position = position;
-            go.transform.rotation = rotation;
+            networkObject.transform.position = position;
+            networkObject.transform.rotation = rotation;
 
             return networkObject;
         }
 
         /// <summary>
         /// Registers all objects in <see cref="PooledPrefabsList"/> to the cache.
         /// </summary>
-        public void InitializePool()
+        void InitializePool()
         {
-            if (m_HasInitialized) return;
```

Comment:
As part of this PR, I removed this field which wasn't necessary. Doing so, and making the InitializePool and ClearPool methods private and only called in OnNetworkSpawn and OnNetworkDespawn respectively, fixes the issue described in #733 and should make this more easily reusable in other projects. Even though it was not necessary in this specific use-case, I think it is cleaner that way regardless